core/src/main/java/org/elasticsearch/action/support/ChildTaskActionRequest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.support;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.tasks.ChildTask;
+import org.elasticsearch.tasks.Task;
+
+import java.io.IOException;
+
+/**
+ * Base class for action requests that can have associated child tasks
+ */
+public abstract class ChildTaskActionRequest<T extends ActionRequest> extends ActionRequest<T> {
+
+    private String parentTaskNode;
+
+    private long parentTaskId;
+
+    protected ChildTaskActionRequest() {
+
+    }
+
+    protected ChildTaskActionRequest(ActionRequest parentTaskRequest) {
+        super(parentTaskRequest);
+    }
+
+    public void setParentTask(String parentTaskNode, long parentTaskId) {
+        this.parentTaskNode = parentTaskNode;
+        this.parentTaskId = parentTaskId;
+    }
+
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        super.readFrom(in);
+        parentTaskNode = in.readOptionalString();
+        parentTaskId = in.readLong();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeOptionalString(parentTaskNode);
+        out.writeLong(parentTaskId);
+    }
+
+    @Override
+    public Task createTask(long id, String type, String action) {
+        return new ChildTask(id, type, action, this::getDescription, parentTaskNode, parentTaskId);
```

Comment:
`this:getDescription` is tricky. I implemented it that way because I want a request to be able to lazily describe itself. In order to do it lazily we need to have a pointer to the request and therefore potentially hold the request in memory longer than needed. Alternatively, we can ask a request to describe itself ahead of time, but it will be wasteful (since most likely nobody will ask for this description). So, in the former case, we will have to make sure that generation of the description is very light-weight operation, which might reduce the ability to identify exactly which task is currently running. Not really sure which solution is better.